hatch/api_client.py

```diff
@@ -0,0 +1,53 @@
+import httpx
+from fastapi import HTTPException, Response
+
+from hatch import config
+
+
+# using a module level client should keep the connection open to job-server
+client = httpx.Client(
+    base_url=config.API_SERVER, headers={"Authorization": config.BACKEND_TOKEN}
+)
+
+
+def create_release(workspace, release, user):
+    """API call to job server to create a release."""
+    response = client.post(
+        url=f"/api/v2/releases/workspace/{workspace}",
+        content=release.json(),
+        headers={"OS-User": user},
+    )
+    if response.status_code != 201:
+        raise proxy_httpx_error(response)
+
+    return proxy_httpx_response(response)
+
+
+def _proxy_headers(orig_headers):
+    headers = orig_headers.copy()
+    for header in ["Connection", "Server"]:
+        headers.pop(header, None)
+    headers["Via"] = config.SERVER_HOST
+    return headers
+
+
+def proxy_httpx_response(response):
+    """Take an upstream httpx response and convert to a proxied FastAPI Response."""
+    return Response(
+        status_code=response.status_code,
+        content=response.content,
+        headers=_proxy_headers(response.headers),
+    )
+
+
+def proxy_httpx_error(response):
+    """Take an upstream httpx response and convert to a proxied FastAPI HTTPException."""
+    try:
+        detail = response.json()["detail"]
+    except Exception:
+        detail = response.content
+    return HTTPException(
```

Comment:
This is somewhat pedantic (it's important to note that I resisted the urge to make a pydantic joke here), but could you add a line above the return? It will seaparate the return block from the try/except block which merged together for me on first read!

Reply:
sure, good point!